test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+function createCreateQuery($tableName, $dataTypes, $colNames, $colNamesAE, $slength)
+{
+    $query = "CREATE TABLE ".$tableName." (\n    ";
+    foreach ($dataTypes as $type) {
+
+        if (dataTypeIsString($type)) {
+            $query = $query.$colNames[$type]." ".$type."(".$slength."), \n    ";
+            $query = $query.$colNamesAE[$type]." ".$type."(".$slength."), \n    ";
+        } else {
+            $query = $query.$colNames[$type]." ".$type.", \n    ";
+            $query = $query.$colNamesAE[$type]." ".$type.", \n    ";
+        }
+    }
+
+    $query = substr($query, 0, -7)."\n)";
+
+    return $query;
+}
+
+// The ALTER TABLE query encrypts columns. Each ALTER COLUMN directive must
+// be preceded by ALTER TABLE
+function constructAlterQuery($tableName, $colNames, $dataTypes, $key, $encryptionType, $slength)
+{
+    $query = '';
+    foreach ($dataTypes as $dataType) {
+
+        $plength = dataTypeIsString($dataType) ? "(".$slength.")" : "";
+        $collate = dataTypeNeedsCollate($dataType) ? " COLLATE Latin1_General_BIN2" : "";
+        $query = $query." ALTER TABLE [dbo].[".$tableName."]
+                          ALTER COLUMN [".$colNames[$dataType]."] ".$dataType.$plength." ".$collate."
+                          ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = [".$key."], ENCRYPTION_TYPE = ".$encryptionType.", ALGORITHM = 'AEAD_AES_256_CBC_HMAC_SHA_256') NOT NULL
+                          WITH
+                          (ONLINE = ON);";
+    }
+
+    $query = $query." ALTER DATABASE SCOPED CONFIGURATION CLEAR PROCEDURE_CACHE;";
+
+    return $query;
+}
+
+// This CREATE TABLE query creates a table with two columns for
+// each data type side by side, one plaintext and one encrypted
+function createAECreateQuery($tableName, $dataTypes, $colNames, $colNamesAE, $slength, $key, $encryptionType)
+{
+    $query = "CREATE TABLE ".$tableName." (\n    ";
+
+    foreach ($dataTypes as $type) {
+
+        $collate = dataTypeNeedsCollate($type) ? " COLLATE Latin1_General_BIN2" : "";
+
+        if (dataTypeIsString($type)) {
+            $query = $query.$colNames[$type]." ".$type."(".$slength.") NULL, \n    ";
+            $query = $query.$colNamesAE[$type]." ".$type."(".$slength.") \n    ";
+            $query = $query."  ".$collate." ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = [".$key."], ENCRYPTION_TYPE = ".$encryptionType.", ALGORITHM = 'AEAD_AES_256_CBC_HMAC_SHA_256') NULL,\n    ";
+        } else {
+            $query = $query.$colNames[$type]." ".$type." NULL, \n    ";
+            $query = $query.$colNamesAE[$type]." ".$type." \n    ";
+            $query = $query."  ".$collate." ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = [".$key."], ENCRYPTION_TYPE = ".$encryptionType.", ALGORITHM = 'AEAD_AES_256_CBC_HMAC_SHA_256') NULL,\n    ";
+        }
+    }
+
+    $query = substr($query, 0, -6)."\n)";
+
+    return $query;
+}
+
+// The INSERT query for the table
+function formulateSetupQuery($tableName, &$dataTypes, &$colNames, &$colNamesAE)
+{
```

Comment:
"SetupQuery" actually means the "InsertQuery" right? Again, use "construct" rather than "formulate" to be consistent with the above functions